Editor/EditorCore/StripProBuilderScripts.cs

```diff
@@ -80,6 +80,9 @@ public static void DoStrip(ProBuilderMesh pb)
 
                 EditorUtility.SynchronizeWithMeshFilter(pb);
 
+                if (go.GetComponent<PolyShape>())
+                    DestroyImmediate(go.GetComponent<PolyShape>());
```

Comment:
You should use a variable to skip searching for the component twice. Something like this instead:
```
var polyShape = go.GetComponent<PolyShape>();
if (polyShape != null)
   DestroyImmediate(polyShape);
```

In this case, we also have a quick version of this that doesn't allocate if the component doesn't exist:
```
if (go.TryGetComponent(out PolyShape polyShape)
    DestroyImmediate(polyShape);
```
You can find the doc here: https://docs.unity3d.com/ScriptReference/GameObject.TryGetComponent.html

Reply:
Since all the function was made this way I changed it in other parts too to make it more uniform and efficient :)